### PR TITLE
[ui] Move asset catalog / search

### PR DESCRIPTION
**Pull request by clairelin135**
More details here: https://github.com/dagster-io/internal/pull/8853

### CODE REVIEW

**Comment by clairelin135** [comment]
* **#20627** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/20627?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @clairelin135 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: js_modules/dagster-ui/packages/ui-core/src/search/BuildAssetSearchResults.tsx [hunk 1]
```diff
@@ -0,0 +1,137 @@
+import {COMMON_COLLATOR} from '../app/Util';
```
Comment (salazarm) [inline js_modules/dagster-ui/packages/ui-core/src/search/BuildAssetSearchResults.tsx:1]: Why is this file here?

Comment (clairelin135) [reply js_modules/dagster-ui/packages/ui-core/src/search/BuildAssetSearchResults.tsx:1]: This is the logic to build the asset filter search results, which are constructed in global search

Comment (salazarm) [reply js_modules/dagster-ui/packages/ui-core/src/search/BuildAssetSearchResults.tsx:1]: I think we should find a way not to leak this code into OSS since it's for a cloud specific feature but we can look into that later.

Comment (clairelin135) [reply js_modules/dagster-ui/packages/ui-core/src/search/BuildAssetSearchResults.tsx:1]: sounds good

**Review by salazarm — APPROVED** [review]
